### PR TITLE
Update deploy to use OCI as distribution mechanism

### PR DESCRIPTION
**Pull request by radu-matei**

### CODE REVIEW

File: src/commands/deploy.rs [hunk 1]
```diff
@@ -393,48 +397,50 @@ impl DeployCommand {
...
+                    + "-"
+                    + &buildinfo
+            }
+            None => cloud_registry_url.domain().unwrap().to_owned() + "/" + &application.info.name,
         };
 
-        let bindle_id = spin_bindle::prepare_bindle(&self.app()?, buildinfo, dest_dir)
-            .await
-            .map_err(crate::wrap_prepare_bindle_error)?;
+        println!("Uploading {} to Fermyon Cloud...", reference);
```
Comment (bacongobbler) [inline src/commands/deploy.rs:429]: Would you mind removing the registry URL here? I don't think there's any action we want users to take with this URL. e.g. they can't use `spin registry push` or `spin registry pull` with this URL IIRC.

Comment (radu-matei) [reply src/commands/deploy.rs:429]: That's a good point. I wasn't sure it should be there to begin with, so I'll remove it.

**Comment by vdice** [comment]
Closing in favor of https://github.com/fermyon/cloud-plugin/pull/57, updated with the latest changes from `main`.

File: src/commands/deploy.rs (hunk 1 above)
Comment (vdice) [reply src/commands/deploy.rs:429]: This has been updated in https://github.com/fermyon/cloud-plugin/pull/57/commits/dd19eec9cc4d98147e2fab6121065101923188e7

**Pull request closed without merging**